### PR TITLE
Check for a custom css file to help customization of instances

### PR DESCRIPTION
**Pull request by Naouak**
People are customizing their instance by modifying directly files from the git.

This pull request permit them to create a `custom.scss` file that will be loaded automatically instead of `application.scss`.

People will be able to rewrite completely the default css or modify variables easily with this system.

Can someone check my work? I'm not a rails dev and I did this with a lot of trials and errors, I'm not sure it's the best way to do it.

### CODE REVIEW

File: app/helpers/style_helper.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module StyleHelper
+  def stylesheet_for_layout
+    if asset_exist?'custom.css'
+      'custom'
+    else
+      'application'
+    end
```
Comment (expenses) [inline app/helpers/style_helper.rb]: `asset_exist? 'custom.css' ? 'custom' : 'application'` might be simpler here. Otherwise, looking good.

Comment (expenses) [inline app/helpers/style_helper.rb]: Didn't mean to review >_<

File: config/initializers/assets.rb [hunk 2]
```diff
@@ -9,4 +9,6 @@
 # Precompile additional assets.
 # application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
 Rails.application.config.assets.precompile += %w(application_public.js)
+Rails.application.config.assets.precompile += %w( custom.css )
```
Comment (Mandragora2) [inline config/initializers/assets.rb]: what happend if file does not exist ?

File: docs/Running-Mastodon/Customizing.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+`custom.scss` file :
+
+````scss
+$color4: #d3d900; // vibrant
+
+@import 'application';
+````
+
+Don't forget to recompile your assets and restart mastodon(if you didn't have a `custom.scss` file before) 
+to see the changes.
```
Comment (Mandragora2) [inline docs/Running-Mastodon/Customizing.md]: 👍 for the doc

File: app/helpers/style_helper.rb (hunk 1 above)
Comment (Naouak) [inline app/helpers/style_helper.rb]: I'm hoping that this helper will get more complicated in later versions (theme-list support? user setting?... etc.) so I think ternary operator may be a bit too much.

**Review by Mandragora2 — APPROVED** [review]

File: config/initializers/assets.rb (hunk 2 above)
Comment (Naouak) [inline config/initializers/assets.rb]: It doesn't do anything. No error reported when custom.scss/custom.css is not there.

Comment (Gargron) [inline config/initializers/assets.rb]: I think you should move custom.css to the line above (`%w(a b)` creates an array `[a, b]`, it's exactly what this is for)

File: app/helpers/style_helper.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module StyleHelper
+  def stylesheet_for_layout
+    if asset_exist?'custom.css'
```
Comment (Gargron) [inline app/helpers/style_helper.rb]: Space after ? please!

File: docs/Running-Mastodon/Customizing.md (hunk 3 above)
Comment (Gargron) [inline docs/Running-Mastodon/Customizing.md]: This doc should move to tootsuite/documentation please!

**Review by Gargron — CHANGES_REQUESTED** [review]
Great idea, just a few minor nitpicks!

**Comment by Naouak** [comment]
I did all the requested changes. I will do a pull request for the removed documentation file when this pull request is accepted,

**Review by Gargron — APPROVED** [review]

**Pull request merged**